ydb/src/load_balancer.rs

```diff
@@ -1,16 +1,27 @@
-use crate::discovery::{Discovery, DiscoveryState};
+use crate::discovery::{Discovery, DiscoveryState, NodeInfo};
 use crate::errors::*;
 use http::Uri;
+use itertools::Itertools;
+use rand::thread_rng;
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+use tokio::runtime::Handle;
+use tokio::sync::mpsc;
+use tokio::task::JoinSet;
+use tokio_util::sync::CancellationToken;
 
 use crate::grpc_wrapper::raw_services::Service;
 use crate::waiter::{Waiter, WaiterImpl};
+use std::collections::HashMap;
+use std::net::ToSocketAddrs;
 use std::sync::{Arc, RwLock};
 use tokio::sync::watch::Receiver;
 
 #[mockall::automock]
 pub(crate) trait LoadBalancer: Send + Sync + Waiter {
     fn endpoint(&self, service: Service) -> YdbResult<Uri>;
     fn set_discovery_state(&mut self, discovery_state: &Arc<DiscoveryState>) -> YdbResult<()>;
+    fn set_config(&mut self, config: BalancerConfig) -> YdbResult<()>;
```

Comment:
The config of balancer is static info, it doesn't need changed in runtime and no need the method in the create.
Try to create balancer with config at start.

It allow to implement different config for different balancer, no need to think about possibility to change config on the fly, etc, no need return errors from static balancer and handle the error.